hexrd/ui/interactive_template.py

```diff
@@ -41,20 +74,17 @@ def rotate_shape(self, angle):
         self.rotate_template(self.shape.xy, angle)
         self.redraw()
 
-    def create_shape(self, module, file_name, det, instr):
+    def create_polygon(self, verts, **kwargs):
         self.complete = False
-        with resource_loader.resource_path(module, file_name) as f:
-            data = np.loadtxt(f)
-        verts = self.panels['default'].cartToPixel(data)
-        verts[:, [0, 1]] = verts[:, [1, 0]]
-        self.shape = patches.Polygon(verts, fill=False, lw=1, color='cyan')
+        self.shape = patches.Polygon(verts, **kwargs)
         if has_nan(verts):
             # This template contains more than one polygon and the last point
             # should not be connected to the first. See Tardis IP for example.
             self.shape.set_closed(False)
-        self.shape_styles.append({'line': '-', 'width': 1, 'color': 'cyan'})
-        self.update_position(instr, det)
-        self.connect_translate_rotate()
+        self.shape_styles.append(kwargs)
+        self.update_position()
+        if not self.static_mode:
+            self.connect_translate_rotate()
```

Comment:
You could potentially put the `if self.static_mode` check inside the `self.connect_translate_rotate()` function as well. But you can decide where it goes.

Reply:
Done